packages/core/strapi/lib/services/entity-service/index.js

```diff
@@ -261,8 +261,9 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
       return null;
     }
 
-    await db.query(uid).delete({ where: { id: entityToDelete.id } });
+    // Delete entity components first, and then the entity
     await deleteComponents(uid, entityToDelete);
+    await db.query(uid).delete({ where: { id: entityToDelete.id } });
```

Comment:
Why did you change the order? It was not working before?

Reply:
`deleteComponents` was querying for the uid entity:
```js
// In deleteComponents:
      const value = await strapi.query(uid).load(entityToDelete, attributeName);
```

the entity was already deleted at the time `deleteComponents` was executed.